jetty9/src/main/java/com/thoughtworks/go/server/util/Jetty9Request.java

```diff
@@ -18,10 +18,16 @@
 
 import org.eclipse.jetty.server.Request;
 
+import javax.servlet.ServletRequestWrapper;
+
 public class Jetty9Request implements ServletRequest {
     private Request request;
 
     public Jetty9Request(javax.servlet.ServletRequest request) {
+        while (request instanceof ServletRequestWrapper) {
```

Comment:
The jetty `Request` instance gets wrapped with `ServletRequestWrapper` implementations by several filters, since the other methods in this class require access to the underlying jetty `Request` we have to unwrap it completely — hence the `while`.